pytorch_lightning/trainer/supporters.py

```diff
@@ -150,7 +263,7 @@ def add(self, predictions):
     def to_disk(self) -> None:
         """Write predictions to file(s).
         """
-        for filepath, predictions in self.predictions.items():
+        for filepath, predictions in self._legacy_predictions.items():
```

Comment:
that is a bit confusing so I do use `attach_predictions` nothing will be saved as `_legacy_predictions` is still empty?

Reply:
I kept previous way to store predictions for BC.